src/librustdoc/test.rs

```diff
@@ -410,12 +450,28 @@ impl Collector {
             current_header: None,
             cratename: cratename,
             opts: opts,
+            code_blocks: block_lines,
+            filename: filename,
+        }
+    }
+
+    fn get_line_from_key(&mut self, key: &String) -> u32 {
+        let (line, need_removal) = if let Some(l) = self.code_blocks.get_mut(key) {
+            let need_removal = l.len() > 1;
+            (l.pop().unwrap_or(1), need_removal)
+        } else {
+            return 1;
+        };
+        if need_removal {
+            self.code_blocks.remove(key);
         }
+        line
```

Comment:
As a precision, rustdoc always did like this so I just took the same mechanism for the block code id.